### PR TITLE
vimc-4028 Python client for Montagu API

### PR DESCRIPTION
**Pull request by EmmaLRussell**
This initial version just authenticates with Montagu and gets the bearer token, which will be used by the task queue to authenticate with OrderlyWeb. Sample endpoint also included to check can get data.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+language: python
+python: 3.6
+
+env:
+  - DOCKER_COMPOSE_VERSION=1.26.0
+
+before_install:
+  - sudo rm /usr/local/bin/docker-compose
```
Comment (EmmaLRussell) [inline .travis.yml:8]: The default version of docker compose is the one with the random strings appended to containers, hence installing a newer version here.

File: src/montagu_api.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    def __init__(self, base_url, username, password):
+        self.base_url = base_url
+        auth_url = self.build_url('authenticate')
+        data = 'grant_type=client_credentials'
+        auth = (username, password)
+        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
+        response = requests.post(auth_url, data=data, auth=auth,
+                                 headers=headers)
+        if response.status_code != 200:
+            msg = 'Unexpected status code: {}. Unable to authenticate.'
```
Comment (richfitz) [inline src/montagu_api.py:14]: Do you want to start on the good foot and test these two lines?

**Review by richfitz — CHANGES_REQUESTED** [review]
This looks super tidy - but should we get a test on the unhappy path there?

File: src/montagu_api.py (hunk 2 above)
Comment (EmmaLRussell) [reply src/montagu_api.py:14]: It was actually a slightly different 2 lines, but point taken!

Comment (richfitz) [reply src/montagu_api.py:14]: oops sorry

**Review by richfitz — APPROVED** [review]

**Pull request merged**